recipes-nodes/render-config/fetch-config.sh

```diff
@@ -12,23 +12,38 @@
 set -e
 
 INIT_CONFIG_FILE="/etc/init-config.json"
-TEMPLATED_CONFIG_FILES="/etc/td-agent-bit/td-agent-bit.conf /etc/process-exporter/process-exporter.yaml /etc/prometheus/prometheus.yml /etc/rbuilder.config /etc/rclone.conf /etc/orderflow-proxy.conf"
+TEMPLATED_CONFIG_FILES="/etc/td-agent-bit/td-agent-bit.conf /etc/process-exporter/process-exporter.yaml /etc/prometheus/prometheus.yml /etc/rbuilder.config /etc/rclone.conf /etc/orderflow-proxy.conf /etc/system-api/systemapi-config.toml"
+SYSTEM_API_FIFO=/var/volatile/system-api.fifo
+
+log() {
+    if [ -p $SYSTEM_API_FIFO ]; then
+        date_log() {
+            echo -n ""
+        }
+    else
+        date_log() {
+            echo -n "$(date): "
+        }
+    fi
+    echo "$(date_log)$1" | tee -a $SYSTEM_API_FIFO
```

Comment:
I have two suggestions here.
You can probably extract the date_log() as its own function and have the if statement in it instead.
something like
date_log() {
if [ -p $SYSTEM_API_FIFO ]; then
            echo -n ""
    else
            echo -n "$(date): "
    fi
}

Then use you can use it directly in the log() function.

The other thing is that wouldn't the log then create the log SYSTEM_API_FIFO even if it doesnt exist with the `tee -a $SYSTEM_API_FIFO` or would it cause the init script to fail if the file is not found?

Reply:
1. `date_log()` is only useful within the `log()` function itself. I don't think it is beneficial to move `date_log()` outside.
2. `tee -a` will append to a file/pipe. If the file does not exists, `tee` will create it. What is your concern here?

Reply:
i'm not a big fan of this `log` function, it's hard to understand and the nested functions don't help.

imo better simplify, and always add the date prefix? that'll simplify the function to:

```
log() {
    echo "$(date)$1" | tee -a $SYSTEM_API_FIFO
}
```

Reply:
If we always append the timestamp after System API starts timestamps will be duplicated. The function supports both variants: before and after System API starts

Reply:
IIRC, the System API filters those message that are coming with timestamp appended to them. So no duplication will happen here

Reply:
> If we always append the timestamp after System API starts timestamps will be duplicated. The function supports both variants: before and after System API starts

yeah i understand what the goal is here. it's just not very pretty and a bit of extra magic happening, that makes the scripts harder to understand. I'm not against that, and feel free to proceed -- just wanted to mention that i'm not a big fan of these complex log functions in several places, and would prefer duplicated timestamps instead. but not strongly held, and am happy to go with whatever you feel is best overall

Reply:
I agree that log function looks cumbersome. If I break it down into two distinct functions it will still be a a few lines of code (agree, maybe a bit more readable). If I do as Chris suggested and simplify it to a single line then we sacrifice the logs readability by duplicating the timestamps when the system boots.

I'd say we bite the bullet and leave it as is for the current iteration. I suggest we address it in the next iteration with a working Syslog (not present in the v1 image) and exposing its contents through System API. This should both simplify the log functions everywhere (or maybe even remove completely) and do not require splitting System API into two components where one responsible for logs and the other for the rest of the functionality.